File: Codeforce/2023/RPC10/Jampacked.py
n,k=map(int,input().split())
if(n%k==0):
    print(k)
elif(k>=n):
    print(n)
elif(k==1):
    print(k)
else:
    cajas=(n//k)+1
    respuesta = n//cajas
    print(respuesta)

File: Codeforce/2023/RPC10/Jampacked.cpp
#include<iostream>

using namespace std;

int main(){
    long long n,k;
    cin>>n>>k;
    if(n%k==0){
        cout<<k;
    }
    else if(k>=n){
        cout<<n;
    }
    else if(k==1){
        cout<<k;
    }
    else{
    long long cajas=(n/k)+1;
    long long respuesta = n/cajas;
    cout<<respuesta;
    }
    return 0;
}